src/dotnet-grpc/Commands/CommandBase.cs

```diff
@@ -90,19 +93,66 @@ public Services ResolveServices(Services services)
             return Services.Client;
         }
 
-        public void EnsureNugetPackages(Services services)
+        public async Task EnsureNugetPackagesAsync(Services services)
         {
+            var packageVersions = await ResolvePackageVersions();
+
             Debug.Assert(services != Services.Default);
 
             foreach (var dependency in GetType().Assembly.GetCustomAttributes<GrpcDependencyAttribute>())
             {
+                // Check if the dependency is applicable for this service type
                 if (dependency.ApplicableServices.Split(';').Any(s => string.Equals(s, services.ToString(), StringComparison.OrdinalIgnoreCase)))
                 {
-                    AddNugetPackage(dependency.Name, dependency.Version, dependency.PrivateAssets);
+                    // Check if the dependency is applicable to this SDK type
+                    if (dependency.ApplicableToWeb == null || string.Equals(dependency.ApplicableToWeb, IsUsingWebSdk.ToString(), StringComparison.OrdinalIgnoreCase))
+                    {
+                        // Use the version specified from the remote file before falling back the packaged versions
+                        var packageVersion = packageVersions?.GetValueOrDefault(dependency.Name) ?? dependency.Version;
+                        AddNugetPackage(dependency.Name, packageVersion, dependency.PrivateAssets);
+                    }
                 }
             }
         }
 
+        private async Task<Dictionary<string, string>?> ResolvePackageVersions()
+        {
+            /* Example Json content
+             {
+              "Version" : "1.0",
+              "Packages"  :  {
+                "Microsoft.Azure.SignalR": "1.1.0-preview1-10442",
+                "Grpc.AspNetCore.Server": "0.1.22-pre2",
+                "Grpc.Net.ClientFactory": "0.1.22-pre2",
+                "Google.Protobuf": "3.8.0",
+                "Grpc.Tools": "1.22.0",
+                "NSwag.ApiDescription.Client": "13.0.3",
+                "Microsoft.Extensions.ApiDescription.Client": "0.3.0-preview7.19365.7",
+                "Newtonsoft.Json": "12.0.2"
+              }
+            }*/
+            try
+            {
+                using var packageVersionStream = await _httpClient.GetStreamAsync(PackageVersionUrl);
+                using var packageVersionDocument = await JsonDocument.ParseAsync(packageVersionStream);
+                var packageVersionsElement = packageVersionDocument.RootElement.GetProperty("Packages");
+                var packageVersionsDictionary = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+                foreach (var packageVersion in packageVersionsElement.EnumerateObject())
+                {
+                    packageVersionsDictionary[packageVersion.Name] = packageVersion.Value.GetString();
+                }
+
+                return packageVersionsDictionary;
+            }
+            catch
```

Comment:
Should probably limit the type of exception here to `JSONParseException` or whatever it's called if the intent is to fail gracefully if the link doesn't point to json.